django/db/models/fields/generated.py

```diff
@@ -30,6 +30,9 @@ def __init__(self, *, expression, output_field, db_persist=None, **kwargs):
         self.expression = expression
         self.output_field = output_field
         self.db_persist = db_persist
+
+        if kwargs.get("choices", None) is None and output_field.choices is not None:
+            kwargs["choices"] = output_field.choices
```

Comment:
This would allow two different ways to do one thing. Ideally we should only allow one. Would it make more sense to limiting this to just `output_field.choices` as the way to set this? There are other parameters to `GeneratedField` that aren't allowed already, such as `default`. Maybe `choices` should be disallowed too.

However, if we do allow multiple sources for choices, then should we include that in the documentation? Otherwise the developer would have to read the code to understand how one of these are being selected.

If multiple sources are provided, we may want to use the `NOT_PROVIDED` pattern instead of relying on `None`. This prevents fewer misunderstandings in what the developer wants to happen versus what actually happens.